### ISSUE TITLE
cards are broken in NODE_ENV=production builds

### ISSUE THREAD

**Issue by marusak** [martinpitt/performance-graphs#10]
![bah](https://user-images.githubusercontent.com/12330670/91827881-ba3ee400-ec3f-11ea-8927-7e8d4b7d65c3.png)

zoom in/out in the browser and sometimes this happens. We had similar issue on the overview page but it was somehow addressed.

**Comment by martinpitt** [comment]
https://garrett.github.io/cockpit-mockweb/metrics/current/ has the same problem. However, Overview page doesn't -- it has lots of complex `@media` specific CSS, but it seems we still need it for some bits at least.

**Comment by martinpitt** [comment]
@marusak: I sometimes get that effect as well, and then the cards are wrong in *every* resolution. Sometimes it seems my built webpack gets in a bad state where it disrespects the scss. When I clean/rebuild the tree, everything is back to normal, and the cards look fine in all resolutions for me.

Do you still get this error with current master? If so, can you check precise steps/window widths when that happens?

**Comment by marusak** [comment]
Ok, I've been trying to reproduce this and cannot find any reproducer.

**Comment by martinpitt** [comment]
[still happens](https://logs.cockpit-project.org/logs/pull-29-20200907-093514-bf0ffa29-fedora-33/TestApplication-testCurrentMetrics-fedora-33-127.0.0.2-2201-FAIL.png)

**Comment by martinpitt** [comment]
I noticed that I can reproduce this when installing from built rpm, and sometimes even with `make`, but never with `make watch`. After some fiddling, the crucial difference is `NODE_ENV=production`, which is set automatically when doing `make vm` as part of `make dist-gzip`. @KKoukiou , does that ring a bell by any chance? I think we had something similar in Cockpit, where we had to add extra `include`s into our .scss for some things?

**Comment by martinpitt** [comment]
Interesting -- so in debug mode, this good rule wins:
```css
.current-metrics {
    --card-width: 22rem;
    --pf-l-gallery--GridTemplateColumns: repeat(auto-fill, minmax(var(--card-width), 1fr));
}
```

and shadows the default rule (striked through in CSS inspector):
```css
.pf-l-gallery {
    --pf-l-gallery--m-gutter--GridGap: var(--pf-global--gutter);
    --pf-l-gallery--GridTemplateColumns: repeat(auto-fill, minmax(250px, 1fr));
    --pf-l-gallery--GridTemplateRows: auto;
    display: grid;
    grid-template-columns: var(--pf-l-gallery--GridTemplateColumns);
    grid-template-rows: var(--pf-l-gallery--GridTemplateRows);
}
```
but in production mode, it's the opposite; when I untick the default rule in the inspector, our rule takes over, and it looks right.

Adding `!important` to our definition works fine. I also tried this, hoping that would get the scoping better:
```diff
--- src/app.scss
+++ src/app.scss
@@ -4,8 +4,10 @@
 
 // FIXME: More styles from ct-system-overview
 .current-metrics {
-  --card-width: 22rem;
-  --pf-l-gallery--GridTemplateColumns: repeat(auto-fill, minmax(var(--card-width), 1fr));
+  .pf-l-gallery {
+      --card-width: 22rem;
+      --pf-l-gallery--GridTemplateColumns: repeat(auto-fill, minmax(var(--card-width), 1fr));
+  }
 
   .pf-c-card {
       &__title {
```
and interestingly this works even worse -- now our rule doesn't appear in the inspector at all any more.

It really feels like some kind of css inclusion ordering issue -- PF's should be included first, and our's as the  last one. Our app.jsx already does `import './app.scss';` as the very last `import` statement, so that can't be it.

Our Overview page uses the same code, without any apparent extra hacks, and that isn't affected.

@garrett, does that ring any bell for you?

**Comment by martinpitt** [comment]
Notes from IRC:
 * The root cause is the dist/index.css ordering problem -- our own app CSS should never come before any included/module/PF CSS.
 * We shouldn't make our CSS selectors more specific. That would help with this particular case, but leave open the trap for the next time.
 * We could try to `@import Gallery.scss` in our css, i. e. import the modules whoses variables you override. That results in putting  the gallery css before our app css (what we want), but keeps the original one as a duplicate after our app css; so the net result is just a bigger, but still broken css. Something like [sass-import-once](https://github.com/at-import/node-sass-import-once) may help, but we don't use that anywhere else.
 * I tried moving around the scss imports, but no change -- the generated dist/index.css stubbornly refuses to change to a proper ordering that matches the `@import` order.

**Comment by martinpitt** [comment]
Finally found a fix, in PR #32
